File: exercises/python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):

        self.matrix = []

        # convert rows to 2D list
        matrix_rows = matrix_string.split('\n')
        for matrix_row in matrix_rows:
            # split matrix_row into a list of integers
            matrix_row_list = list(map( int, matrix_row.split(' ') ))

            # Add matrix_row_list as a nested list in matrix
            self.matrix.append(matrix_row_list)


    def row(self, index):
        return self.matrix[index - 1]

    def column(self, index):
        return [matrix_row[index - 1] for matrix_row in self.matrix[:]]

File: exercises/typescript/matrix/matrix.ts
class Matrix {

  public rows:    number[][]
  public columns: number[][]

  constructor( matrix_string: string ) {

    // Populate rows 2D array
    this.rows = matrix_string
      .split("\n")
      .map( (row_string: string) =>
        row_string
        .split(" ")
        .map( (number_as_string: string): number =>
          Number(number_as_string)
        )
      );

    // double loop to populate columns array
    this.columns = [];
    // This should work for rectangular matrices.
    // iterate over column numbers in first row of array
    for ( var col_index: number = 0; col_index < this.rows[0].length ; col_index++ ) {
      // iterate over row numbers in array
      this.columns[col_index] = [];
      for ( var row_index: number = 0; row_index < this.rows.length ; row_index++ ) {
        // switch col_index <-> row_index between rows[][] and columns[][]
        this.columns[col_index][row_index] = this.rows[row_index][col_index];
      }
    }
  }
}

export default Matrix
